internal/controller/warehouses/warehouses.go

```diff
@@ -290,10 +302,18 @@ func (r *reconciler) getLatestFreightFromRepos(
 ) (*kargoapi.Freight, error) {
 	logger := logging.LoggerFromContext(ctx)
 
+	repoCommitMappings := map[string]string{}
+	if warehouse.Status.LastFreight != nil {
+		for _, commit := range warehouse.Status.LastFreight.Commits {
+			repoCommitMappings[commit.RepoURL+"#"+commit.Branch] = commit.ID
+		}
+	}
+
 	selectedCommits, err := r.selectCommitsFn(
 		ctx,
 		warehouse.Namespace,
 		warehouse.Spec.Subscriptions,
+		repoCommitMappings,
```

Comment:
I think things would be tidier and more testable overall if we passed the `warehouse.Status.LastFreight` here and let `selectCommits()` build the map.

Reply:
refactored:
https://github.com/akuity/kargo/pull/1477/commits/b8667c672b0f4819c7522de56e0859e11d6595ba